.github/workflows/test.yml

```diff
@@ -109,3 +109,60 @@ jobs:
           bin_name: test_bin
           # Test setting just workdir.
           workdir: testdata
+  
+  action-test-smoke-test-pass:
+    runs-on: ubuntu-latest
+    env:
+      OVERRIDE_TARBALL_NAME: action-test-smoke-test-pass.docker.tar
+    needs:
+      - action-test-prep
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Build
+        uses: ./
+        with:
+          version: 1.0.0
+          target: default
+          arch: amd64
+          tags: |
+            ${{env.TAG_PREFIX}}/action-test-smoke-test-pass:tag1
+            ${{env.TAG_PREFIX}}/action-test-smoke-test-pass:tag2
+          zip_artifact_name: test_bin.zip
+          bin_name: test_bin
+          dockerfile: testdata/Dockerfile
+          smoke_test: |
+            docker run "${IMAGE_NAME}" /bin/"${BIN_NAME}"
+
+  action-test-smoke-test-fail:
+    runs-on: ubuntu-latest
+    env:
+      OVERRIDE_TARBALL_NAME: action-test-smoke-test-fail.docker.tar
+    needs:
+      - action-test-prep
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Build
+        id: docker-build
+        uses: ./
+        with:
+          version: 1.0.0
+          target: default
+          arch: amd64
+          tags: |
+            ${{env.TAG_PREFIX}}/action-test-smoke-test-fail:tag1
+            ${{env.TAG_PREFIX}}/action-test-smoke-test-fail:tag2
+          zip_artifact_name: test_bin.zip
+          bin_name: test_bin
+          dockerfile: testdata/Dockerfile
+          smoke_test: |
+            docker run "${IMAGE_NAME}"
+        continue-on-error: true
+      - name: Check smoke test fail
+        env:
+          TEST_RESULT: ${{ steps.docker-build.outcome }}
+        run: | 
+          if [ "${TEST_RESULT}" == 'success' ]; then
+            exit 1
+          fi
```

Comment:
I'm really happy with how this turned out, thank you for adding these tests @jeanneryan! Your idea to add some verbose output when the test passes or fails sounds good to me as well.